### PR TITLE
Fix null pointer errors in protocol v5 when using prepared statements

### LINKED ISSUE #96: Null pointer exceptions with prepared statements on protocol v5
This happens because `ResultMetadataId` is always set to `null`  on the `Prepared` responses.

### PR DESCRIPTION
**Pull request by joao-r-reis**
Also added some logic to the `FrameEncoder` so that these errors are caught, logged and a server error is returned to the client.

I was able to reproduce the bug with the test that I added.

I also fixed some unit tests that were failing on Windows because of the line separator.

Fixes #96

### CODE REVIEW

**Comment by joao-r-reis** [comment]
After this change all C# simulacron tests pass locally with protocol v5

File: native-server/src/test/java/com/datastax/oss/simulacron/server/BoundNodeTest.java [hunk 1]
```diff
@@ -17,9 +17,7 @@
 
 import static org.assertj.core.api.Assertions.assertThat;
 
-import com.datastax.oss.protocol.internal.Frame;
-import com.datastax.oss.protocol.internal.Message;
-import com.datastax.oss.protocol.internal.ProtocolConstants;
+import com.datastax.oss.protocol.internal.*;
```
[the remark this answers is not in the document]
Comment (joao-r-reis) [reply native-server/src/test/java/com/datastax/oss/simulacron/server/BoundNodeTest.java:20]: hmm IntelliJ added this automatically, should I change it ? I was building and running locally without errors

File: native-server/src/main/java/com/datastax/oss/simulacron/server/FrameEncoder.java [hunk 2]
```diff
@@ -17,21 +17,36 @@
...
   }
 
   @Override
-  protected void encode(ChannelHandlerContext ctx, Frame msg, List<Object> out) throws Exception {
-    out.add(frameCodec.encode(msg));
+  protected void encode(ChannelHandlerContext ctx, Frame msg, List<Object> out) {
+    try {
+      out.add(frameCodec.encode(msg));
+    } catch (Throwable t) {
+      logger.error("Exception while encoding a frame. Returning a server error instead.", t);
```
[the remark this answers is not in the document]
Comment (joao-r-reis) [reply native-server/src/main/java/com/datastax/oss/simulacron/server/FrameEncoder.java:45]: Yeah, I think there's other places where this happens but we can fix them as we run into them

File: native-server/src/test/java/com/datastax/oss/simulacron/server/BoundNodeTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (adutra) [reply native-server/src/test/java/com/datastax/oss/simulacron/server/BoundNodeTest.java:20]: The [contribution guidelines](https://github.com/datastax/simulacron/blob/master/CONTRIBUTING.md#code-formatting) forbid wildcard imports. Let's try to stick to the conventions.

[the remark this answers is not in the document]
Comment (joao-r-reis) [reply native-server/src/test/java/com/datastax/oss/simulacron/server/BoundNodeTest.java]: My IDE wasn't configured correctly, oops. Just pushed a commit to fix the imports.

**Pull request merged**